File: 刷题/golang/1. 两数之和.go
func twoSum(nums []int, target int) []int {
	mp := map[int]int{}
	for i, x := range nums {
		if p, ok := mp[target-x]; ok {
			return []int{p, i}
		}
		mp[x] = i
	}
	return nil
}

File: 刷题/cpp/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hashTable;
        for(auto i=0;i<nums.size();i++){
            if(hashTable.find(target-nums[i])!=hashTable.end()){
                return {hashTable[target-nums[i]],i};
            }
            hashTable[nums[i]]=i;
        }
        return {};
    }
};
// @lc code=end

